src/ResourceManager/DataFactories/Commands.DataFactories/Gateway/GetAzureDataFactoryGatewayAuthKeyCommand.cs

```diff
@@ -0,0 +1,60 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.DataFactories.Models;
+using Microsoft.Azure.Commands.DataFactories.Properties;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using System.Security.Permissions;
+
+namespace Microsoft.Azure.Commands.DataFactories
+{
+    [Cmdlet(VerbsCommon.Get, Constants.GatewayAuthKey, DefaultParameterSetName = ByFactoryName), OutputType(typeof(PSDataFactoryGatewayAuthKey))]
+    public class GetAzureDataFactoryGatewayAuthKeyCommand : DataFactoryBaseCmdlet
+    {
+        [Parameter(ParameterSetName = ByFactoryObject, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The data factory object.")]
+        public PSDataFactory DataFactory { get; set; }
```

Comment:
@nicolaszhang for new cmdlets, we are trying to ensure that they follow the guidelines and patterns established by the PowerShell Core team, even if that means they go against a design previously used for cmdlets. Even if this parameter doesn't share the same name as the one from the cmdlet you mentioned, it can still have an alias to `DataFactory`, so users will still be able to use that name if they don't want to use `InputObject`.